processor/src/main/java/org/mapstruct/ap/internal/util/Message.java

```diff
@@ -107,7 +105,13 @@
     INHERITCONFIGURATION_DUPLICATE_MATCHES( "Given name \"%s\" matches several candidate methods: %s." ),
     INHERITCONFIGURATION_NO_NAME_MATCH( "Given name \"%s\" does not match the only candidate. Did you mean: \"%s\"." ),
     INHERITCONFIGURATION_MULTIPLE_PROTOTYPE_METHODS_MATCH( "More than one configuration prototype method is applicable. Use @InheritConfiguration to select one of them explicitly: %s." ),
-    INHERITCONFIGURATION_CYCLE( "Cycle detected while evaluating inherited configurations. Inheritance path: %s" );
+    INHERITCONFIGURATION_CYCLE( "Cycle detected while evaluating inherited configurations. Inheritance path: %s" ),
+
+    VALUEMAPPING_DUPLICATE_SOURCE( "Source value mapping: \"%s\" cannot be mapped more than once." ),
+    VALUEMAPPING_INCORRECT_NAME_BASE_CONTINUEATION( "Source == \"$\" can only be used in combination with target == \"$\"." ),
```

Comment:
Use `source = "$"` instead of `==` as it is displayed at the location of the assignment.